common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java

```diff
@@ -156,6 +193,20 @@ protected void handleMessage(
       Map<String, String[]> localDirs = blockManager.getLocalDirs(msg.appId, msg.execIds);
       callback.onSuccess(new LocalDirsForExecutors(localDirs).toByteBuffer());
 
+    } else if (msgObj instanceof FinalizeShuffleMerge) {
+      final Timer.Context responseDelayContext =
+          metrics.finalizeShuffleMergeLatencyMillis.time();
+      FinalizeShuffleMerge msg = (FinalizeShuffleMerge) msgObj;
+      try {
+        checkAuth(client, msg.appId);
+        MergeStatuses statuses = mergeManager.finalizeShuffleMerge(msg);
+        callback.onSuccess(statuses.toByteBuffer());
+      } catch(IOException e) {
+        throw new RuntimeException(String.format("Error while finalizing shuffle merge "
+            + "for application %s shuffle %d", msg.appId, msg.shuffleId));
```

Comment:
I think we should respond to DAGScheduler with an error instead of throwing an exception here so that the finalizing future at DAGScheduler can complete quickly.

Reply:
By throwing a RuntimeException here, it will be caught by `TransportRequestHandler`, which will then respond a `RPCFailure` back to client.
Would this satisfy the requirement?

Reply:
Yeah, sounds good.